### PR TITLE
Fix deprecated option in sphinx that causes warning treated as error in travis

### LINKED ISSUE #2401: travis env is in bad shape -- all fail due to sphinx config
While testing #2399 saw them all failing... a sample env fails with

```
  warnings.warn('Matplotlib is building the font cache using fc-list. This may take a moment.')
loading pickled environment... failed: ('__init__() takes exactly 4 arguments (1 given)', <class 'sphinx.environment.BuildEnvironment'>, ())
Warning, treated as error:
WARNING: latex_use_modeindex is deprecated. Use latex_domain_indices instead.
```

### PR DESCRIPTION
**Pull request by sciunto**
See #2371

Closes #2401

Ref:
* https://github.com/sphinx-doc/sphinx/blob/master/sphinx/builders/latex.py#L253
* http://www.sphinx-doc.org/en/1.4.8/config.html#confval-latex_domain_indices

### CODE REVIEW

**Comment by sciunto** [comment]
Looks like there is several evolutions in sphinx.

Right now, I have this message
```
Exception occurred:

  File "/home/travis/build/scikit-image/scikit-image/doc/source/../ext/sphinx_gallery/docs_resolv.py", line 48, in _get_data

    with open(url, 'r') as fid:

IOError: [Errno 2] No such file or directory: '/home/travis/build/scikit-image/scikit-image/doc/build/html/api/skimage.color.rst.html'

The full traceback has been saved in /tmp/sphinx-err-yAUWXd.log, if you want to report the issue to the developers.

Please also report this if it was a user error, so that a better error message can be provided next time.

A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

It deserves some investigations... ping @sphinx-gallery (not sure if this is relevant: https://github.com/sphinx-gallery/sphinx-gallery/issues/58)

**Comment by JDWarner** [comment]
Sphinx broke this in version 1.5, their logic for warning checking has (had) a typo.  I reported it here and it's now fixed in master: https://github.com/sphinx-doc/sphinx/issues/3204

The workaround is either to use Sphinx from master until they release a version with the fix, or allow Travis to pass with Sphinx errors in the interim.

**Comment by sciunto** [comment]
Thanks @JDWarner for your comment but actually, I already went through that one, no? (And i didn't noticed the typo). My problem is more with sphinx-gallery now.

**Comment by sciunto** [comment]
I updated sphinx-gallery, a bug fix release for sphinx 1.5.

**Comment by emmanuelle** [comment]
Thanks @sciunto ! For the sphinx-gallery files, did you copy them from the latest sphinx-gallery release (or from master), or is there some divergence between the files in scikit-image and the ones in sphinx-gallery ?

**Comment by sciunto** [comment]
@emmanuelle I copied them, and I didn't modify any of them.

**Comment by emmanuelle** [comment]
OK, great. At some point we can consider adding sphinx-gallery as a dependency, but there's no hurry to do it. Two questions:
- any idea why Travis still fails?
- with the new sphinx-gallery, did you build the gallery to check that it still looks good?

**Comment by sciunto** [comment]
@emmanuelle Did you read the ticket I opened upstream? Before going into more investigations, we were wondering if there was modifications that you brought in our copy of sphinx gallery that were not included upstream? I guess you would never have done that, except perhaps by mistake...

**Comment by sciunto** [comment]
PS: the problem also exists on my desktop. Nevertheless, the gallery looks good as it happens at the very end for the generation of hyperlinks.

**Comment by emmanuelle** [comment]
@sciunto is it https://github.com/sphinx-gallery/sphinx-gallery/issues/184? When I ported the gallery to sphinx-gallery (#2078) I had to modify at least one file but I back-ported it to sphinx-gallery (https://github.com/sphinx-gallery/sphinx-gallery/pull/119, which has been merged).

**Comment by emmanuelle** [comment]
As suggested by @JDWarner should we allow Travis to pass with sphinx warnings until the problem is solved? It's blocking a number of PRs in the meantime...

**Comment by sciunto** [comment]
Thanks for the clarifications @emmanuelle 

We can do that. I don't know where is this option to do not treat warning as errors. If you give me a hint, I can do it in this PR. At least, this PR solves problems related to our configuration and the latest sphinx evolutions. It could be merged imho. For the rest, upstream is currently looking at it. I can open another ticket to follow the progress.

**Comment by sciunto** [comment]
A patch is in the pipes...

**Comment by sciunto** [comment]
@emmanuelle @soupault The build goes fine on my side. Ready to go!

**Comment by Borda** [comment]
it seems ready to merge, right? :)

**Comment by sciunto** [comment]
Let's ping @scikit-image/core to see if someone can press the green button :)

File: doc/ext/sphinx_gallery/__init__.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 """
 import os
-__version__ = '0.1.2'
+__version__ = '0.1.7'
```
Comment (soupault) [inline doc/ext/sphinx_gallery/__init__.py]: If I understood correctly, the update in this PR contains sphinx-gallery 0.1.7+https://github.com/sphinx-gallery/sphinx-gallery/pull/190 .
If it is indeed so, I'd suggest either to wait for that patch being merged, or to note somewhere the details regarding the version used.

**Comment by sciunto** [comment]
@soupault My plan is to re-update sphinx-gallery before the release... I open a ticket for that, assigned to myself.

**Comment by sciunto** [comment]
The idea is to be get travis back to work for other PRs.

File: doc/ext/sphinx_gallery/__init__.py (hunk 1 above)
Comment (emmanuelle) [inline doc/ext/sphinx_gallery/__init__.py]: I think it's more urgent to have Travis passing again.

**Comment by emmanuelle** [comment]
@sciunto in addition to the open ticket, can you add a line in TODO.txt as well?

**Comment by sciunto** [comment]
Oki, it's done :)

**Comment by emmanuelle** [comment]
Merging, thanks!

**Comment by soupault** [comment]
LGTM as well.

**Pull request merged**